### ISSUE TITLE
Fabio v1.5.2, GO v1.8.3 Socket Issue in LX

### ISSUE THREAD

**Issue by Smithx10** [joyent/illumos-joyent#147]
While running the latest version of [fabio](https://github.com/fabiolb/fabio/releases/download/v1.5.2/fabio-1.5.2-go1.8.3-linux_amd64) (Fabio v1.5.2, GO v1.8.3) in LX I ran into the following error while starting.
`
2017/08/08 03:34:56 [FATAL] ui: set tcp 192.168.131.127:9998->192.168.131.127:63643: setsockopt: invalid argument
`

I took an strace -f fabio and uploaded it to https://manta.tritonhost.com/smith/public/pastes/fabiostrace-lx.  Leftwing from the #SmartOS felt this was the relevant information from that strace.

```
LeftWing> So, these look relevant:
4:30 AM [pid 13104] setsockopt(7, SOL_TCP, TCP_KEEPINTVL, [180], 4 <unfinished ...>
4:30 AM [pid 13104] <... setsockopt resumed> )  = -1 EINVAL (Invalid argument)
4:30 AM [pid 13104] write(2, "2017/08/08 03:23:48 [FATAL] ui: "..., 1142017/08/08 03:23:48 [FATAL] ui: set tcp 192.168.131.127:9998->192.168.131.127:49894: setsockopt: invalid argument
4:31 AM (I clipped out some less important context)
4:31 AM Either we don't support TCP_KEEPINTVL or we don't support the value 180 for it, I guess?
```

**Comment by jjelinek** [comment]
I filed OS-6276 for this. We do support TCP_KEEPINTVL, but it appears that we're missing the code to set that option.

**Comment by jjelinek** [comment]
Actually, the setting code is there after all, but we might be out of the valid range.

**Comment by jjelinek** [comment]
I tried a quick experiment locally and I could set the values to 10-60, but not higher. We'll have to check into this.

**Comment by Smithx10** [comment]
@jjelinek Any update on this?  Is this issue more complex than initially thought?

**Comment by jjelinek** [comment]
Actually, it just slipped off my list because of higher priority work, but I should be able to get back to it hopefully during this week. Sorry about the delay.

**Comment by Smithx10** [comment]
@jjelinek  No issue, It's not a pressing matter.  Thanks for the update.  Maybe one day I'll be capable of fixing a bug like this one and not have to rely on the kernel gods :P

**Comment by jjelinek** [comment]
This is fixed with commit 1ee89f3.
